setup.py

```diff
@@ -85,6 +85,7 @@
     'pytest_codeblocks>=0.16.1,<0.18',
     'pytest-cov>=4,<7',
     'pyright==1.1.256',
+    'pydantic==1.10.10',
```

Comment:
pinning pydantic is not the right answer here. we shouldn't have any dependency on pydantic or deepspeed and need to figure out what is causing that